### PR TITLE
JDK-8303476: Add the runtime version in the release file of a JDK image

### PR DESCRIPTION
**Pull request by mlchung**
`$JAVA_HOME/release` file currently includes `JAVA_VERSION`  which is the version 
number plus the pre-release identifier for example "21-ea".    The build number 
and additional build information are useful in identifying the information 
about a JDK image but not available.  This proposes to add a new entry in the release file:
```
JAVA_RUNTIME_VERSION="$(VSTR)"
```
where `$(VSTR)` is the string represented by `Runtime::version()`.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change requires CSR request [JDK-8303479](https://bugs.openjdk.org/browse/JDK-8303479) to be approved
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issues
 * [JDK-8303476](https://bugs.openjdk.org/browse/JDK-8303476): Add the runtime version in the release file of a JDK image
 * [JDK-8303479](https://bugs.openjdk.org/browse/JDK-8303479): Add JAVA_RUNTIME_VERSION in the release file of a JDK image (**CSR**)

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12813/head:pull/12813` \
`$ git checkout pull/12813`

Update a local copy of the PR: \
`$ git checkout pull/12813` \
`$ git pull https://git.openjdk.org/jdk pull/12813/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12813`

View PR using the GUI difftool: \
`$ git pr show -t 12813`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12813.diff">https://git.openjdk.org/jdk/pull/12813.diff</a>

</details>

### CODE REVIEW

**Comment by erikj79** [comment]
> `$JAVA_HOME/release` file currently includes `JAVA_VERSION` which is the version
> number plus the pre-release identifier for example "21-ea".

I can't actually find that in the current release file, but that shouldn't prevent us from adding the full version string.

File: test/jdk/build/releaseFile/CheckReleaseFile.java [hunk 1]
```diff
@@ -78,6 +80,12 @@ private void readFile(String fileName, boolean isOpenJDK) {
                     implementor = readIn;
                     continue;
                 }
+
+                // grab IMPLEMENTOR line
```
Comment (erikj79) [inline test/jdk/build/releaseFile/CheckReleaseFile.java:84]: This comment looks out of place.

**Comment by mlchung** [comment]
> I can't actually find that in the current release file, but that shouldn't prevent us from adding the full version string.

`JAVA_VERSION` is added by jlink and not in ReleaseFile.gmk.

**Comment by erikj79** [comment]
> > I can't actually find that in the current release file, but that shouldn't prevent us from adding the full version string.
> 
> `JAVA_VERSION` is added by jlink and not in ReleaseFile.gmk.

Ah, that explains it, if I look in images/jdk/release it's present.

**Review by erikj79 — APPROVED** [review]

**Comment by mlchung** [comment]
Can you also review CSR?  thanks.

**Pull request closed without merging**